awscli/clidriver.py

```diff
@@ -314,28 +320,51 @@ def main(self, args=None):
                 'CLI_VERSION', self.session.user_agent(), 'CLI')
             HISTORY_RECORDER.record('CLI_ARGUMENTS', args, 'CLI')
             return command_table[parsed_args.command](remaining, parsed_args)
+        except (BotocoreParamValidationError, ParamValidationError) as e:
+            # RC 252 represents that the command failed to parse or failed
+            # client side validation at the botocore level.
+            LOG.debug("Client side parameter validation failed", exc_info=True)
+            write_exception(e, outfile=get_stderr_text_writer())
+            return 252
         except UnknownArgumentError as e:
             sys.stderr.write("usage: %s\n" % USAGE)
             sys.stderr.write(str(e))
             sys.stderr.write("\n")
-            return 255
+            return 252
+        except ConfigurationError as e:
+            # RC 253 represents that the command may be syntatically correct
```

Comment:
We should be documenting these in the topic guide on rc's: https://github.com/aws/aws-cli/blob/v2/awscli/topics/return-codes.rst

Reply:
Updated it to reflect the new return codes. We should probably figure something out for codes 1-127. Right now it looks like we're trying to overload the general topic guide with some command specific definitions (the S3 ones) which probably won't work out long term.